### PR TITLE
feat: add new samples, update linters

### PR DESCRIPTION
**Pull request by sethvargo**
Fills in missing samples and updates existing samples to the new format. This also updates the linter to match newer repos like Secret Manager.

### CODE REVIEW

**Comment by sethvargo** [comment]
Ping @bcoe

**Comment by sethvargo** [comment]
I'm not really sure how to make the linter happy? It seems to be complaining about the generated TypeScript files.

File: .eslintrc.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+extends:
+  - 'eslint:recommended'
+  - 'plugin:node/recommended'
+  - prettier
```
Comment (bcoe) [inline .eslintrc.yml:5]: these linting templates are managed centrally by the code generator now, and we just took on a ton of work to make all the libraries consistent [here](https://github.com/googleapis/nodejs-kms/pull/309).

tldr; I don't think we should need to update any of this linting configuration manually.

File: .eslintignore [hunk 2]
```diff
@@ -1,6 +1,5 @@
 **/node_modules
-**/coverage
-test/fixtures
+src/**/doc/*
```
Comment (bcoe) [inline .eslintignore:2]: likewise this is auto-generated.

File: .eslintrc.yml (hunk 1 above)
Comment (sethvargo) [reply .eslintrc.yml]: reverted

File: .eslintignore (hunk 2 above)
Comment (sethvargo) [reply .eslintignore]: reverted
